src/app/public/modules/i18n/currency/currency-format.service.ts

```diff
@@ -0,0 +1,118 @@
+import { getCurrencySymbol } from '@angular/common';
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { map, take } from 'rxjs/operators';
+
+import { SkyCurrencyFormat } from './currency-format';
+import { SkyAppLocaleProvider } from '../locale-provider';
+
+type IsoCodeAndLocale = { isoCode: string; locale: string };
+type CurrencyFormatParts = {
+  symbol: string;
+  symbolLocation: 'p' | 's';
+  decimalCharacter: string;
+  groupCharacter: string;
+};
+
+/**
+ * Service to get a currency's format given an iso code.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class SkyCurrencyFormatService {
+  constructor(private localeProvider: SkyAppLocaleProvider) { }
+
+  /**
+   * Gets the currency formatting.
+   * @param params optional params object
+   * @param params.isoCode the ISO 4217 Code. Defaults to `USD`.
+   * @param params.locale the locale.
+   */
+  public getCurrencyFormat(params?: Partial<IsoCodeAndLocale>): SkyCurrencyFormat {
+    const isoCode = params?.isoCode ?? 'USD';
+    const locale = params?.locale ?? 'en-US';
+    const formatter = new Intl.NumberFormat(locale, { style: 'currency', currency: isoCode });
+
+    const resolvedOptions: Intl.ResolvedNumberFormatOptions = formatter.resolvedOptions();
+    const currencyCode = resolvedOptions.currency ?? isoCode;
+    const parts = this.formatToParts(formatter, currencyCode, locale);
+
+    return {
+      locale: locale,
+      isoCode: currencyCode,
+      symbol: parts.symbol,
+      symbolLocation: parts.symbolLocation,
+      decimalCharacter: parts.decimalCharacter,
+      groupCharacter: parts.groupCharacter,
+      precision: resolvedOptions.maximumFractionDigits
+    };
+  }
+
+  /**
+   * Gets an observable of the currency formatting.
+   * @param params optional params object
+   * @param params.isoCode the ISO 4217 Code. Defaults to `USD`.
+   * @param params.locale the locale. Defaults to `SkyAppLocaleProvider.getLocaleInfo`.
+   */
+  public getCurrencyFormatAsync(params?: Partial<IsoCodeAndLocale>): Observable<SkyCurrencyFormat> {
```

Comment:
SkyAppLocaleProvider returns an observable (bc users can override it) so I think providing a non-observable-locking call could be valuable.

Defintiely open to opionions though.